admin/class-primary-term-admin.php

```diff
@@ -200,10 +245,13 @@ private function map_taxonomies_for_js( $taxonomy ) {
 		}
 
 		return array(
-			'title'   => $taxonomy->labels->singular_name,
-			'name'    => $taxonomy->name,
-			'primary' => $primary_term,
-			'terms'   => array_map( array( $this, 'map_terms_for_js' ), get_terms( $taxonomy->name ) ),
+			'title'          => $taxonomy->labels->singular_name,
+			'name'           => $taxonomy->name,
+			'primary'        => $primary_term,
+			'singular_label' => $taxonomy->labels->singular_name,
+			'field_id'       => $this->generate_field_id( $taxonomy->name ),
+			'rest_base'      => ( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxonomy->name,
+			'terms'          => array_map( array( $this, 'map_terms_for_js' ), get_terms( $taxonomy->name ) ),
```

Comment:
This is too hard to read, please put this in a method. I see this was the old situation, but boyscouting this is a good practice.